src/app/geo-accession-download/geo-accession-download.component.ts

```diff
@@ -0,0 +1,44 @@
+import {Component, Input} from '@angular/core';
+import {LoaderService} from "@shared/services/loader.service";
+import {SaveFileService} from "@shared/services/save-file.service";
+import {BrokerService} from "@shared/services/broker.service";
+import {AlertService} from "@shared/services/alert.service";
+
+@Component({
+  selector: 'app-geo-accession-download',
+  templateUrl: './geo-accession-download.component.html',
+  styleUrls: ['./geo-accession-download.component.css']
+})
+
+export class GeoAccessionDownloadComponent {
+  @Input() geoAccession;
+
+  constructor(
+    private loaderService: LoaderService,
+    private saveFileService: SaveFileService,
+    private brokerService: BrokerService,
+    private alertService: AlertService,
+  ) { }
+
+
+  downloadSpreadsheetUsingGeo() {
+    this.loaderService.display(true, 'This may take a moment. Please wait...')
+    const geo_accession = this.geoAccession;
+    if (geo_accession) {
+      this.brokerService.downloadSpreadsheetUsingGeo(geo_accession)
+        .subscribe(response => {
+            const filename = response['filename'];
+            const blob = new Blob([response['data']]);
+            this.saveFileService.saveFile(blob, filename);
+            this.loaderService.hide();
+          },
+          error => {
+            console.log(error)
+            this.alertService.error('Unable to download spreadsheet using this GEO accession.' +
+              ' Please retry again.', error.message );
+            this.loaderService.hide();
+          })
+    }
+
```

Comment:
You need to hide the loader service here. If there's no geo accession the loading spinnner will never go away

Reply:
Or actually, if you do my above comment you can just move the return to the first line of the function so that the loader is never shown if there's not an accession. Could be a good idea to add a log if there's no accession

Reply:
there's a check in the html, so the download function will never get called if the geo accession isn't there.

I've moved the `loaderService.display` bit to inside the if block

Reply:
Please add a log. If there's a reason to add the check for geo accession in the function, there's a reason to log it. In both cases, it's because this function could be used differently in the future

Reply:
added a log in case the geo accession isn't there